### PR TITLE
chore(ci): run react-native tests on next-release

### PR DESCRIPTION
**Pull request by wlee221**
#### Description of changes

With `next-release/main` containing RN changes now, we need to run react-native unit tests on that branch. 

For that reason, this PR creates a `test-next-release.yml` that runs `reusable-e2e@next-release/main` instead of `reusable-e2e@main`. This is similar to how we have `test-rna.yml`, `test-in-app-messaging.yml`, etc.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/test-next-release-prs.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Description: This workflow runs unit + e2e tests on PRs targeting
```
Comment (wlee221) [inline .github/workflows/test-next-release-prs.yml:1]: Copy and pasted from `test-rwa.yml`. In the future, we can definitely improve this exprience of having to copy paste workflows for every non-trivial feature branch. I'll take a look after all the feature works are done.

**Review by calebpollman — APPROVED** [review]

**Comment by 0618** [comment]
merge to `main`?

File: .github/workflows/test-next-release-prs.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        if: always() && github.event.label.name == 'run-tests'
+        uses: actions/github-script@0.3.0
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const { issue: { number: issue_number }, repo: { owner, repo } } = context;
+            const label = 'run-tests';
+            github.issues.removeLabel({ owner, repo, issue_number, name: label });
+  test:
+    uses: aws-amplify/amplify-ui/.github/workflows/reusable-e2e.yml@next-release/main
```
Comment (wlee221) [inline .github/workflows/test-next-release-prs.yml:43]: Actual fix here -- we're targeting `@next-release/main` instead of `main` in this workflow.

**Comment by wlee221** [comment]
> merge to main?

yes, I should've explained it. I'll merge it to main first, then sync this to `next-release/main`. `pull_request_target` is [weird](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target), and runs on base branch instead of the head branch. So I think it's generally better to just put them in `main` as source of truth, and have them transfer over to downstream feature branches.

**Review by ErikCH — APPROVED** [review]

**Comment by wlee221** [comment]
I'll go ahead and merge this in at 6pm PT, unless there are objections.

**Pull request merged**